doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        if (matches.Size() > 0)
+        {
+            m_installPackageOperation = InstallPackage(matches.GetAt(0).CatalogPackage());
+            UpdateUIForInstall(m_installPackageOperation, installButton, cancelButton, progressBar, statusText);
+        }
+    }
+
+```
+
+## 3.3.1 Cancel
+
+The async operation must be stored, or the install code must wait on an event that can be triggered.
+
+```c++ (C++ish pseudocode)
+    void MainPage::CancelButtonClickHandler(IInspectable const&, RoutedEventArgs const&)
+    {
+        if (m_installPackageOperation)
+        {
+            m_installPackageOperation.Cancel();
+        }
+    }
+```
+
+## 3.3.2. Cancel
+
+Cancel the async operation
+
+```c++ (C++ish pseudocode)
+    IAsyncAction CancelInstall(
+        std::wstring installAppId)
+    {
+        co_await winrt::resume_background();
+        // Creation of the AppInstaller has to use CoCreateInstance rather than normal winrt initialization since it's created 
+        // by an out of proc com server.
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog windowsCatalog{ appInstaller.GetAppCatalog(PredefinedAppCatalog::OpenWindowsCatalog) };
+        windowsCatalog.OpenAsync().get();
+        GetCompositeAppCatalogOptions getCompositeAppCatalogOptions = CreateGetCompositeAppCatalogOptions();
+        getCompositeAppCatalogOptions.Catalogs().Append(windowsCatalog);
+        getCompositeAppCatalogOptions.LocalAppCatalog(LocalAppCatalog::InstallingPackages);
+        // Specify that the search behavior is to only query for local packages.
+        // Since the local catalog that is open is InstallingPackages, this will only find a result if installAppId is 
+        // currently installing.
+        getCompositeAppCatalogOptions.CompositeSearchBehavior(CompositeSearchBehavior::InstallingPackages);
+        AppCatalog compositeCatalog{ appInstaller.GetCompositeAppCatalog(getCompositeAppCatalogOptions) };
+        co_await compositeCatalog.OpenAsync();
+
+        FindPackagesOptions findPackagesOptions = CreateFindPackagesOptions();
+        PackageMatchFilter filter;
+        filter.IsAdditive(true);
+        filter.Field(PackageMatchField::Id);
+        filter.Type(MatchType::Exact);
+        filter.Value(installAppId);
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ compositeCatalog.FindPackagesAsync(findPackagesOptions).get() };
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        CatalogPackage package = matches.GetAt(0).CatalogPackage();
+
+        if (package.IsInstalling())
+        {
+            Windows::Foundation::IAsyncOperationWithProgress<InstallResult, InstallProgress> installOperation = appInstaller.GetInstallProgress(package);
+            installOperation.Cancel();
+        }
+    }
+
+```
+
+## 3.3. Get progress for installing app

```

Comment:
3.4 now (and increment below ones)?

Reply:
If you use Visual Code to edit Markdown files the 'Markdown All in One' extension makes it easy to fixup numbering

* command (<code>ctrl-p</code>) to <code>Add/Update Section Numbers</code>. Rerun whenever adding or changing sections (</code>#</code> prefixed lines)

I also find these helpful

* command (<code>ctrl-p</code>) to <code>Create Table of Contents</code>. A TOC is created at the cursor. This is a static list based on the file's current content. Only need to do once.
* command (<code>ctrl-p</code>) to <code>Update Table of Contents</code>. The TOC created by <code>Create Table of Contents</code> is rewritten based on the file's current content.